src/redprl/pretty.sml

```diff
@@ -1 +1,28 @@
 structure PP = PrettyPrint (AnsiColors)
+
+structure TermPrinter = 
+struct
+  structure Abt = RedPrlAbt
+  structure ShowVar = Abt.Var
+  structure O = RedPrlOpData
+
+  structure UP = UnparseAbt (structure Abt = Abt and Unparse = Unparse)
+
+  open Abt O Unparse
+
+  fun @@ (f, x) = f x
+  infix 0 @@ $ $$ \
+
+  fun notation m = 
+    case Abt.out m of
+      (* A -> B *)
+      MONO DFUN $ [_ \ a, (_, []) \ bx] =>
+        SOME (atom @@ toString a ^ " -> " ^ toString bx)
```

Comment:
Great start! One correction:

The way that ABTs work here is that they have a fixed arity; this means that there is only one correct way to write a DFUN, and it is with a single variable bound; so, this case will never fire. The case below this will fire, though.

If you want to add separate printing for when the bound variable does not occur in b, you could do something like:

```sml
if Var.Ctx.member (varctx bx) x then (* ... *) else (* ... *)
```